exchanges/okx/okx_websocket.go

```diff
@@ -0,0 +1,2155 @@
+func (ok *Okx) TradesSubscription(operation string, assetType asset.Item, pair currency.Pair) (*SubscriptionOperationResponse, error) {
+	return ok.WsChannelSubscription(operation, okxChannelTrades, assetType, pair, false, true)
+}
+
+// EstimatedDeliveryExercisePriceSubscription to subscribe or unsubscribe to "estimated-price" channel to retrieve the estimated delivery/exercise price of FUTURES contracts and OPTION.
+func (ok *Okx) EstimatedDeliveryExercisePriceSubscription(operation string, assetType asset.Item, pair currency.Pair) (*SubscriptionOperationResponse, error) {
+	if !(assetType == asset.Futures || assetType == asset.Option) {
+		return nil, fmt.Errorf("%w, only FUTURES and OPTION asset types are supported", errInvalidInstrumentType)
+	}
+	return ok.WsChannelSubscription(operation, okxChannelEstimatedPrice, assetType, pair, true, true, false)
+}
+
+// MarkPriceSubscription to subscribe or unsubscribe to to "mark-price" to retrieve the mark price. Data will be pushed every 200 ms when the mark price changes, and will be pushed every 10 seconds when the mark price does not change.
+func (ok *Okx) MarkPriceSubscription(operation string, assetType asset.Item, pair currency.Pair) (*SubscriptionOperationResponse, error) {
+	return ok.WsChannelSubscription(operation, okxChannelMarkPrice, assetType, pair, false, true)
+}
+
+// MarkPriceCandlesticksSubscription to subscribe or unsubscribe to "mark-price-candles" channels to retrieve the candlesticks data of the mark price. Data will be pushed every 500 ms.
+func (ok *Okx) MarkPriceCandlesticksSubscription(operation, channel string, assetType asset.Item, pair currency.Pair) (*SubscriptionOperationResponse, error) {
+	if !(channel == okxChannelMarkPriceCandle1Y || channel == okxChannelMarkPriceCandle6M || channel == okxChannelMarkPriceCandle3M || channel == okxChannelMarkPriceCandle1M || channel == okxChannelMarkPriceCandle1W || channel == okxChannelMarkPriceCandle1D || channel == okxChannelMarkPriceCandle2D || channel == okxChannelMarkPriceCandle3D || channel == okxChannelMarkPriceCandle5D || channel == okxChannelMarkPriceCandle12H || channel == okxChannelMarkPriceCandle6H || channel == okxChannelMarkPriceCandle4H || channel == okxChannelMarkPriceCandle2H || channel == okxChannelMarkPriceCandle1H || channel == okxChannelMarkPriceCandle30m || channel == okxChannelMarkPriceCandle15m || channel == okxChannelMarkPriceCandle5m || channel == okxChannelMarkPriceCandle3m || channel == okxChannelMarkPriceCandle1m || channel == okxChannelMarkPriceCandle1Yutc || channel == okxChannelMarkPriceCandle3Mutc || channel == okxChannelMarkPriceCandle1Mutc || channel == okxChannelMarkPriceCandle1Wutc || channel == okxChannelMarkPriceCandle1Dutc || channel == okxChannelMarkPriceCandle2Dutc || channel == okxChannelMarkPriceCandle3Dutc || channel == okxChannelMarkPriceCandle5Dutc || channel == okxChannelMarkPriceCandle12Hutc || channel == okxChannelMarkPriceCandle6Hutc) {
```

Comment:
```suggestion
	if channel != okxChannelMarkPriceCandle1Y && channel != okxChannelMarkPriceCandle6M && channel != okxChannelMarkPriceCandle3M && channel != okxChannelMarkPriceCandle1M && channel != okxChannelMarkPriceCandle1W && channel != okxChannelMarkPriceCandle1D && channel != okxChannelMarkPriceCandle2D && channel != okxChannelMarkPriceCandle3D && channel != okxChannelMarkPriceCandle5D && channel != okxChannelMarkPriceCandle12H && channel != okxChannelMarkPriceCandle6H && channel != okxChannelMarkPriceCandle4H && channel != okxChannelMarkPriceCandle2H && channel != okxChannelMarkPriceCandle1H && channel != okxChannelMarkPriceCandle30m && channel != okxChannelMarkPriceCandle15m && channel != okxChannelMarkPriceCandle5m && channel != okxChannelMarkPriceCandle3m && channel != okxChannelMarkPriceCandle1m && channel != okxChannelMarkPriceCandle1Yutc && channel != okxChannelMarkPriceCandle3Mutc && channel != okxChannelMarkPriceCandle1Mutc && channel != okxChannelMarkPriceCandle1Wutc && channel != okxChannelMarkPriceCandle1Dutc && channel != okxChannelMarkPriceCandle2Dutc && channel != okxChannelMarkPriceCandle3Dutc && channel != okxChannelMarkPriceCandle5Dutc && channel != okxChannelMarkPriceCandle12Hutc && channel != okxChannelMarkPriceCandle6Hutc {
```

This would be much more readable with a map ok check